typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,125 @@
+        for (let fileId in filesToGenerateSet) {
+            let relativeFilepath, found = generatedFileRelativeFilepaths[fileId]; //TODO - does this work (having map values return 2 things??)
+            if (!found) {
+                return [null, new Error(
+                    "No filepath (relative to test volume root) was returned for file '%v', even though we requested it; this is a Kurtosis bug", //TODO - no %v maybe
+                    )];
+            }
+            // let absFilepathOnTestsuite = new URL(this.testVolumeMountpointOnTestsuiteContainer, relativeFilepath); //TODO need to confirm this VS path
+            // let absFilepathOnService = new URL(this.testVolumeMountpointOnServiceContainer, relativeFilepath);
+            let absFilepathOnTestsuite = path.Join(this.testVolumeMountpointOnTestsuiteContainer, relativeFilepath);
+            let absFilepathOnService = path.Join(this.testVolumeMountpointOnServiceContainer, relativeFilepath);
+            let genFilePath = new GeneratedFileFilepaths(); //TODO = this might be a problem in terms of memory (direct or reference), so check
+            genFilePath.absoluteFilepathOnTestsuiteContainer = absFilepathOnTestsuite;
+            genFilePath.absoluteFilepathOnServiceContainer = absFilepathOnService;
+            result[fileId].add(genFilePath); //TODO - = VS add()
+            //}
+        }
+        return [result, null]; //TODO - Design decision, returning multiple values as an array
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public loadStaticFiles(usedStaticFilesSet: [Set<StaticFileID>, Error]): [Map<StaticFileID, string>, Error] { // TODO - change string to StaticFILEID ; (comment) - changed Map into Set
+        let serviceId = this.serviceId;
+        let staticFilesToCopyStringSet = new Map(); //TODO (comment because no Map Type) = map[string]bool{}
+        for (let staticFileId in usedStaticFilesSet) {
+            staticFilesToCopyStringSet[String(staticFileId)] = true;
+        }
+
+        let loadStaticFilesArgs = constructors.newGetLoadStaticFilesArgs(serviceId, staticFilesToCopyStringSet);
+        let request: grpc.requestCallback<bindingsJs.LoadStaticFilesResponse>;
+
+    	let loadStaticFilesResp, err = this.client.loadStaticFiles(loadStaticFilesArgs, request);
+    	if (err != null) {
+    		return [null, new Error ("An error occurred loading the requested static files into the namespace of service '%v'")]; //TODO - remove the erorr, and need to be throwing back the erorr
+    	}
+    	let staticFileAbsFilepathsOnService = new Map(); //TODO (comment) = original type map[StaticFileID]string{}
+    	for (let staticFileId in loadStaticFilesResp.CopiedStaticFileRelativeFilepaths) { //TODO - how to for loop over map keys and values?
+    		let filepathRelativeToExVolRoot = loadStaticFilesResp.CopiedStaticFileRelativeFilepaths[staticFileId];
```

Comment:
Oh hmm, I would also need to fix this indentation, odd that Visual Studio Code is showing me it being perfectly indented